### PR TITLE
upload the smoke testing for locations and attendees route

### PR DESCRIPTION
**Pull request by joseSuper**
Smoke testing of location and attendees route. I had problems with the merge in the fork

### CODE REVIEW

File: tests/Smoke/attendeesRoute.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Smoke testing for attendees Route by: Jose Antonio Cardozo
+ */
+ var init = require('../../init');
+ var config = require(GLOBAL.initialDirectory+'/config/config.json');
+var expect = require('chai').expect;
+var tokenAPI = require(GLOBAL.initialDirectory+config.path.tokenAPI);
+var endPoints = require(GLOBAL.initialDirectory+config.path.endPoints);
+var servicesAPI = require(GLOBAL.initialDirectory+config.path.roomManagerAPI);
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/attendeesRoute.js]: the name of this var should be related to the function (roomManagerAPI)

File: tests/Smoke/attendeesRoute.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+var token = null;
+var url = config.url + endPoints.servicesByIdEndPoint;
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/attendeesRoute.js]: what is the function of this var?

File: tests/Smoke/attendeesRoute.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			});
+	});
+
+	it('Get /services/{servicesId}/attendees?filter=....', function (done) {
+		
+	  servicesAPI
+	  	//get the services id
+		.getwithToken(token,url, function (err,res) {
+			var servicesID = res.body[0]._id;
+			var endPointServicesById = endPointById.replace('{:serviceId}',servicesID);
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/attendeesRoute.js]: we have a method in util/util.js called string replace to do this

File: tests/Smoke/locations.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+	it('GET /locations', function(done) {
+		locationAPI
+			.get(url,function (err,res){
+				expect(res.status).to.equal(config.httpStatus.Ok);
+				done();
+			});
+	});
+
+	it('POST /locations', function(done) {
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/locations.js]: could you please use before and after in your TCs

File: tests/Smoke/locations.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
+/**
+ * Smoke test of location by: Jose Antonio Cardozo
+ */
+var init = require('../../init');
+var config = require(GLOBAL.initialDirectory+'/config/config.json');
+var expect = require('chai').expect;
+var tokenAPI = require(GLOBAL.initialDirectory+config.path.tokenAPI);
+var endPoints = require(GLOBAL.initialDirectory+config.path.endPoints);
+var locationAPI = require(GLOBAL.initialDirectory+config.path.roomManagerAPI);
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/locations.js]: the name of this var should be related at the function

File: util/util.js [hunk 6]
```diff
@@ -22,4 +22,25 @@ var getRandomResourcesJson = function(size)
...
+ * Parameters:
+ *   sizeName        - is the size of name of location
+ *   customNameSize  - is the size of display name
+ *   descriptionSize - is a small description about of location.
+ * Returns:
+ *   return the json with name, custonName and description of location.
+ */
+var generateLocationJson = function (sizeName, customNameSize, descriptionSize) {
+	
+	var locationsJson =
```
Comment (AlejandroJeanCarlo) [inline util/util.js]: this Json should be obtained from a config file

**Comment by joseSuper** [comment]
added the before and after

File: tests/Smoke/locations.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+		it('GET /locations/{:locationId}', function(done) {
+			locationAPI
+				.get(endPointLocationById,function (err,res) {
+					expect(res.status).to.equal(config.httpStatus.Ok);
+					done();
+				});
+		});
+
+		it('PUT /locations/{:locationId}', function (done) {
+			console.log(endPointLocationById);
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/locations.js]: remove this please

**Comment by AlejandroJeanCarlo** [comment]
your code have conflicts! a can not marge your pull request

**Comment by joseSuper** [comment]
ok done.

**Comment by AlejandroJeanCarlo** [comment]
Why this request is closed? You will make a new one?

**Comment by joseSuper** [comment]
Re open

File: tests/Smoke/locations.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+		it('PUT /locations/{:locationId}', function (done) {
+					locationAPI
+						.put(token,endPointLocationById,locationJson,function (err,res) {
+							expect(res.status).to.equal(config.httpStatus.Ok);
+							done()
+						});
+				});
+		});
+
+		it('DELETE /locations/{:locationId}', function (done) {
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/locations.js]: This TC should too use a before, and I think you already have the before please check it

Comment (joseSuper) [inline tests/Smoke/locations.js]: which? the TC delete?, already I have a before on the which is created the location, but also have a after that does deleted of location was created in the before.
and I think that would repeated code

Comment (AlejandroJeanCarlo) [inline tests/Smoke/locations.js]: You are creating a location into the test, the test is focus to delete only

Comment (joseSuper) [inline tests/Smoke/locations.js]: you says that I have create a new describe with a before only for this test?.

Comment (AlejandroJeanCarlo) [inline tests/Smoke/locations.js]: That would be work, also please review de other comments

**Comment by joseSuper** [comment]
all done

File: tests/Smoke/locations.js (hunk 8 above)
Comment (joseSuper) [inline tests/Smoke/locations.js]: all done.

File: tests/Smoke/locations.js [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+var expect = require('chai').expect;
+var tokenAPI = require(GLOBAL.initialDirectory+config.path.tokenAPI);
+var endPoints = require(GLOBAL.initialDirectory+config.path.endPoints);
+var roomManagerAPI = require(GLOBAL.initialDirectory+config.path.roomManagerAPI);
+var util = require(GLOBAL.initialDirectory+config.path.util);
+var locationConfig = require(GLOBAL.initialDirectory+config.path.locationConfig);
+
+//global variables
+var token = null;
+var url = config.url + endPoints.locations;
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/locations.js]: With is the function of this variable?

File: tests/Smoke/attendeesRoute.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+ */
+ var init = require('../../init');
+ var config = require(GLOBAL.initialDirectory+'/config/config.json');
+var expect = require('chai').expect;
+var tokenAPI = require(GLOBAL.initialDirectory+config.path.tokenAPI);
+var endPoints = require(GLOBAL.initialDirectory+config.path.endPoints);
+var roomManagerAPI = require(GLOBAL.initialDirectory+config.path.roomManagerAPI);
+
+var token = null;
+var url = config.url + endPoints.servicesByIdEndPoint;
```
Comment (AlejandroJeanCarlo) [inline tests/Smoke/attendeesRoute.js]: Same

File: util/util.js (hunk 6 above)
Comment (AlejandroJeanCarlo) [inline util/util.js]: This is not done

**Comment by joseSuper** [comment]
changes of name variable done.

**Comment by joseSuper** [comment]
refactor the locationJson

File: tests/CRUD/locations.js [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+			});
+	});
+	
+	it.skip('CRUD of GET /locations', function(done) {
+		roomManagerAPI
+			.get(endPoint,function (err,res){
+				expect(res.status).to.equal(config.httpStatus.Ok);
+				console.log('the size is = ' + res.body.length);
+				expect(res.body).to.not.be.null;
+				//expect(res.body.length).to.be.at.least(1);
```
Comment (AlejandroJeanCarlo) [inline tests/CRUD/locations.js]: Do not should be commented code

File: tests/CRUD/locations.js [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+				// this expect compared to the array of locations are same that the locations that are in the mongoDB, but not yet implemented
+				expect(res.body).to.eql('todas las locations de la base de datos');
```
Comment (AlejandroJeanCarlo) [inline tests/CRUD/locations.js]: Ok is in progress but please to it in English

File: tests/CRUD/locations.js [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+					expect(res.body.path).to.equal(location.path);
+					expect(res.body.name).to.equal(location.name);
+					expect(res.body.customName).to.equal(location.customName);
+					expect(res.body.description).to.equal(location.description);
+					done();
+				});
+		});
+
+		it('CRUD PUT /locations/locationId', function (done) {
+			// generated the changes that want modify in the location.
```
Comment (AlejandroJeanCarlo) [inline tests/CRUD/locations.js]: If the TC is done please remove this comment

File: tests/CRUD/resources.js [hunk 14]
```diff
@@ -59,6 +59,7 @@ describe('CRUD Suite', function () {
 				expect(res.status).to.equal(config.httpStatus.Ok);
 				expect(res.body).to.have.property("name")
 					.and.be.equal(resourceJson.name);
+					console.log(res.body.name)
```
Comment (AlejandroJeanCarlo) [inline tests/CRUD/resources.js]: Remove this line and please do a Doble check before add the files in a commit

File: util/util.js [hunk 15]
```diff
@@ -1,6 +1,7 @@
 // util
 var config = require('../config/config.json')
 var resourceConfig = require('../config/resource.json');
+var locationCongig = require('../config/locations.json');
```
Comment (AlejandroJeanCarlo) [inline util/util.js]: We should to use the root directory with this two lines

**Comment by AlejandroJeanCarlo** [comment]
I put some comments please review those, I will merge your code but please make the changes and commit

**Pull request merged**